compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/AbstractTypeSymbol.java

```diff
@@ -156,7 +157,10 @@ protected List<FunctionSymbol> filterLangLibMethods(List<FunctionSymbol> functio
         List<FunctionSymbol> filteredFunctions = new ArrayList<>();
 
         for (FunctionSymbol function : functions) {
-
+            if (function instanceof BallerinaFunctionSymbol &&
+                    ((BallerinaFunctionSymbol) function).getInternalSymbol().getOrigin() == SymbolOrigin.VIRTUAL) {
+                continue;
+            }
```

Comment:
cc @dulajdilshan

Reply:
How come we didn't come across this previously?

Reply:
Ping on this @chiranSachintha @dulajdilshan

Reply:
Previously we haven't got any values such as this here to do the filtering. Let's better check why this is happening with the new changes. 

However, it would be better to do the filtering in https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/LangLibrary.java

Reply:
The issue arises when using a parameter as the default value for another parameter without creating a new symbol for it. That is not correct. With this I have fixed that, and now we create a new symbol for that. So we need to skip those function as they are created in the desugar level. https://github.com/ballerina-platform/ballerina-lang/blob/4d53eb7dbc56a06765c522063867570b5b3b4585/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureGenerator.java#L606

@dulajdilshan We need to skip them within the `filterLangLibMethods`. Isn't it?

Reply:
How about we filter them in here: https://github.com/ballerina-platform/ballerina-lang/blob/9215217d2b692de0438c8dbe40b6c3619845e387/compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/LangLibrary.java#L179

Reply:
Updated.